pkg/addonmanager/manager.go

```diff
@@ -76,6 +78,11 @@ func (a *addonManager) Start(ctx context.Context) error {
 		return err
 	}
 
+	v1CSRSupported, _, err := utils.IsCSRSupported(kubeClient)
```

Comment:
could we add an additional flag for addon to check this? the v1CSRSupported could be true initially, but the value is set if the flag for the manager is enabled. I am concerned about addition rbac requirement for addon controller.

Reply:
i believe there's no worry in the RBAC compatibility. i was thinking about having a flag to manually switch between enabling/disabling CSR controllers, but by leveraging rest mapper the detection of v1 CSR api can be done completely automatically. so we dont need the flag, wdyt?